pandas/core/internals/managers.py

```diff
@@ -1057,6 +1057,9 @@ def iset(
                     "Shape of new values must be compatible with manager shape"
                 )
 
+            if value.shape[0] > self.shape[0]:
+                raise AssertionError
```

Comment:
This needa a better error message

Reply:
Wondering if we should just be mirroring what numpy does here?

```python
>>> arr = np.zeros((4, 1))
>>> arr2 = np.random.randn(4, 2)
>>> arr[:, 0] = arr2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: could not broadcast input array from shape (4,2) into shape (4,)
```

Somewhat surprised this doesn't happen natrually